src/plasmapy/plasma/material.py

```diff
@@ -0,0 +1,193 @@
+"""
+Definition of the material class and helper functions. The module also
+includes type definitions for stopping and ranging data.
+"""
+
+from collections.abc import Callable
+from functools import cached_property
+from typing import TypedDict
+
+import astropy.units as u
+import numpy as np
+from scipy.interpolate import CubicSpline
+
+from plasmapy.particles.particle_class import Particle
+
+__all__ = [
+    "MaterialSTARData",
+    "Material",
+]
+
+from plasmapy.utils.decorators import validate_quantities
+
+
+class MaterialSTARData(TypedDict):
+    """Type definition for formatting data related to beam-target interactions."""
+
+    incident_particle: Particle
+
+    energies: u.Quantity[u.MeV]
+    stopping_power: u.Quantity[
+        u.MeV * u.cm**2 / u.g
+    ]  # Units of energy per areal density
+    CSDA_range: u.Quantity[u.cm]
```

Comment:
How is CSDA range different than projected range? I thought all the projected range data was using the CDSA approximation.